provider/datagen/src/databake.rs

```diff
@@ -53,14 +55,16 @@ struct ImplData {
 }
 
 impl BakedDataExporter {
-    pub fn new(
-        mod_directory: PathBuf,
-        pretty: bool,
-        insert_feature_gates: bool,
-        use_separate_crates: bool,
-    ) -> Result<Self, DataError> {
-        if mod_directory.exists() {
-            std::fs::remove_dir(&mod_directory)
+    pub fn new(mod_directory: PathBuf, options: BakedOptions) -> Result<Self, DataError> {
+        let BakedOptions {
+            pretty,
+            insert_feature_gates,
+            use_separate_crates,
+            overwrite,
+        } = options;
+
+        if overwrite && mod_directory.exists() {
```

Comment:
If overwrite is false and the dir exists you should still remove it. https://github.com/unicode-org/icu4x/blob/main/provider/fs/src/export/fs_exporter.rs#L96-L100